### ISSUE TITLE
Indication that data is stale?

### ISSUE THREAD

**Issue by quatrano** [apollographql/apollo-client#9549]
**Intended outcome:**

When I pass changed variables to the `useQuery` hook, I'd like some indication that the `data` returned is stale. 

> Stale data is fine, but I'd like some indication that the data is stale. (I see other bug reports where people ask for `data=undefined`, that's **not** what I'm asking for). 

<details><summary>Examples of "some indication" that the data is stale:</summary>

- loading == true
- networkStatus != 7
- variables == [old variables]
- ...anything?
</details>

**Actual outcome:**

Instead, the response is indistinguishable (as far as I can tell) from the case where the data does match the variables.

**How to reproduce the issue:**

https://github.com/quatrano/react-apollo-error-template/pull/1/files
1. click on a person from the list to set the variable for the query
2. click on a different person from the list to change the variable
3. notice 3 renders as a result of action2 - the first is problematic.

<details><summary>console log output from action 2 listed below</summary>

```
BADrender0= {
    "requested": "2",
    "response": "1",
    "loading": false,
    "networkStatus": 7,
    "responseVariables": "2"
}
OKrender1= {
    "requested": "2",
    "response": undefined,
    "loading": true,
    "networkStatus": 2,
    "responseVariables": "2"
}
OKrender3= {
    "requested": "2",
    "response": "2",
    "loading": false,
    "networkStatus": 7,
    "responseVariables": "2"
}
```
</details>

**Versions**

```
  System:
    OS: macOS 11.6
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Browsers:
    Chrome: 99.0.4844.83
    Safari: 15.0
  npmPackages:
    @apollo/client: ^3.5.10 => 3.5.10 
```

**Comment by quatrano** [comment]
I imagine this behavior is intended and allows `useQuery` to accept non-memoized variables and not have to check deep equality during rendering. 

I'd go through the effort of memoizing the variables if that meant the Apollo Client gave me a better response.  Once I do the memoization though, it's not hard to implement the rest on my end. That's probably what you're going to tell me. 🙂

**Comment by dylanwulf** [comment]
This doesn't seem like intentional behavior to me. apollo client should never be returning data from a previous set of variables (that's what `previousData` is for). And memoization of variables isn't supposed to have any different behavior than non-memoized variables. Reverting to apollo client v3.4.17 seems to fix the reproduction, so I think this might be a bug.

**Comment by dylanwulf** [comment]
Looks like this is fixed by #9599

**Comment by bignimbus** [comment]
Thanks all - closing this due to the above but please feel free to let the team know if you need more support 🙏🏻
